sparse/boot/flash-config.sh

```diff
@@ -0,0 +1,39 @@
+VALID_PRODUCTS=(
+@VALID_PRODUCTS@
+)
```

Comment:
Something's wrong while constructing the array in the flash.sh from submodule:
```
    echo ${VALID_PRODUCTS[@]}
    echo ${VALID_PRODUCTS[0]}
```
yields
```
"H4113" "H4133"
"H4113" "H4133"
```
And thus flashing fails:
```
Found H4113, baseband:1311-2918_50.2.A.0.400, bootloader:1310-0301_X_Boot_SDM630_LA3.0_P_38
Found 0 devices: 
Incorrect number of devices connected. Make sure there is exactly one device connected in fastboot mode.
```

My flash.sh contains this after substitution [1]:
`        VALID_PRODUCTS=($(printf %s "-e "H4113" -e "H4133"" | sed 's/-e //g'))`

[1] `sed -e "s/\@DEVICES\@/\-e \"H4113\" -e \"H4133\"/g" -i flash.sh`

Reply:
Looks like IFS is enough to fix (no wonder shellcheck frowned at earlier approach anyway:) https://github.com/koalaman/shellcheck/wiki/SC2207):
`IFS=" " VALID_PRODUCTS=($(printf %s "-e "H4113" -e "H4133"" | sed 's/-e //g'))`